File: c++/problem 1-5.cpp
//Problem #1
#include <iostream>
#include <iomanip>
#include <vector>
using namespace std;

int main() {
    vector<int> nums(10);
    for (int i = 0; i < 10; ++i) {
        cin >> nums[i];
    }
    int m = nums[6];
    double a = m * 2.75;
    cout << "The multiplier is " << m 
         << "! Lulu’s allowance is $ " 
         << fixed << setprecision(2) << a << "!";
    return 0;
}
//Problem #2
#include <iostream>
#include <string>
using namespace std;

int main() {
    int N; 
    cin >> N;
    for (int i = 0; i < N; ++i) {
        int t; string s;
        cin >> t >> s;
        int cnt = 0;
        for (char c : s) {
            if (c >= 'A' && c <= 'Z') ++cnt;
        }
        cout << cnt << "\n";
    }
    return 0;
}
//Problem #3
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    int a, b, c;
    cin >> a >> b >> c;
    double avg = (a + b + c) / 3.0;
    cout << fixed << setprecision(2) << avg;
    return 0;
}
//Problem #4
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int a, b, c;
    cin >> a >> b >> c;
    int orig[3] = {a, b, c};
    int sorted[3] = {a, b, c};
    sort(sorted, sorted + 3);

    cout << sorted[0] << " " << sorted[1] << " " << sorted[2] << "\n";
    cout << orig[0]   << " " << orig[1]   << " " << orig[2];
    return 0;
}
//Problem #5
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    while (cin >> N) {
        vector<int> a(N);
        for (int i = 0; i < N; ++i) {
            cin >> a[i];
        }
        bool first = true;
        for (int i = N - 1; i >= 0; --i) {
            if (a[i] % 2 == 0) {
                if (!first) cout << " ";
                cout << a[i];
                first = false;
            }
        }
        cout << "\n";
    }
    return 0;

File: Fortran/Problem1-5.hs
Problem #1
import Text.Printf (printf)

main :: IO ()
main = do
  line <- getLine
  let nums = map read (words line) :: [Int]
      m    = nums !! 6
      allowance = fromIntegral m * 2.75
  printf "The multiplier is %d! Lulu’s allowance is $ %.2f!" m allowance
Problem #2
import Control.Monad (replicateM_)
import Data.Char (isUpper)

main :: IO ()
main = do
  nLine <- getLine
  let n = read nLine :: Int
  replicateM_ n $ do
    line <- getLine
    let (_:s:_) = words line
        cnt     = length (filter isUpper s)
    print cnt
Problem #3
import Text.Printf (printf)

main :: IO ()
main = do
  line <- getLine
  let [a, b, c] = map read (words line) :: [Double]
      avg       = (a + b + c) / 3
  printf "%.2f" avg
Problem #4
import Data.List (sort)

main :: IO ()
main = do
  line   <- getLine
  let nums   = map read (words line) :: [Int]
      sorted = sort nums
  putStrLn $ unwords (map show sorted)
  putStrLn $ unwords (map show nums)
Problem #5
import Control.Monad (unless)

main :: IO ()
main = do
  contents <- getContents
  mapM_ processLine (lines contents)

processLine :: String -> IO ()
processLine line = do
  let ws      = words line
  case ws of
    (nStr:nums) ->
      let n        = read nStr :: Int
          vals     = take n (map read nums :: [Int])
          evensRev = filter even (reverse vals)
      in putStrLn $ unwords (map show evensRev)
    _ -> return ()
